internal/providers/terraform/provider.go

```diff
@@ -234,14 +230,25 @@ func runPlan(opts *CmdOptions, planFlags string) (string, []byte, error) {
 	args = append(args, flags...)
 	_, err = Cmd(opts, append(args, fmt.Sprintf("-out=%s", f.Name()))...)
 
-	// If the plan returns this error then Terraform is configured with remote execution mode
-	if err != nil && strings.HasPrefix(extractStderr(err), "Error: Saving a generated plan is currently not supported") {
-		log.Info("Continuing with Terraform Remote Execution Mode")
-		config.Environment.TerraformRemoteExecutionModeEnabled = true
-		planJSON, err = runRemotePlan(opts, args)
-	}
-
 	if err != nil {
+		extractedErr := extractStderr(err)
+
+		// If the plan returns this error then Terraform is configured with remote execution mode
+		if strings.HasPrefix(extractedErr, "Error: Saving a generated plan is currently not supported") {
+			log.Info("Continuing with Terraform Remote Execution Mode")
+			config.Environment.TerraformRemoteExecutionModeEnabled = true
+			planJSON, err = runRemotePlan(opts, args)
+		} else if initOnFail == true && (strings.Contains(extractedErr, "Error: Could not load plugin") ||
```

Comment:
```suggestion
		} else if initOnFail && (strings.Contains(extractedErr, "Error: Could not load plugin") ||
```
From CI's output:
 Error: S1002: should omit comparison to bool constant, can be simplified to `initOnFail` (gosimple)